user_coffee_shop_query.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'distance_calculator'
+
+class UserCoffeeShopQuery
+  attr_reader :distances
+
+  def initialize(coffee_shops, user)
+    @coffee_shops = coffee_shops
+    @user = user
+  end
+
+  def show_closest
+    calculate_distances
+    @distances.sort_by { |coffee_shop| coffee_shop[:distance] }.first(3).each do |coffee_shop|
+      puts "#{coffee_shop[:name]},#{coffee_shop[:distance]}"
```

Comment:
The functionality of showing the first 3 items is highly dependent on the I/O method. How would the architecture change if we needed the same logic but export on an endpoint or write to a file?

Reply:
If we we're to send a string to an endpoint, I would first concatenate the whole answer and then send it to the endpoint/write to a file. Of course, a better option would be to send the @distances which could be converted to json, being an array of hashes.